scripts/accounting/seed-expense-categories-from-csv.ts

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env node
+
+import '../../server/env';
+
+import fs from 'fs';
+
+import axios from 'axios';
+import { program } from 'commander';
+import { parse } from 'csv-parse/sync'; // eslint-disable-line
+
+import models, { Op, sequelize } from '../../server/models';
+
+const DRY_RUN = process.env.DRY_RUN !== 'false';
+
+const loadFileContents = async (filePathOrUrl: string): Promise<string> => {
+  if (filePathOrUrl.match(/^https?:\/\//)) {
+    const response = await axios.get(filePathOrUrl);
+    return response.data;
+  } else {
+    const content = await fs.promises.readFile(filePathOrUrl, 'utf8');
+    return content;
+  }
+};
+
+type RunOptions = {
+  verbose: boolean;
+};
+
+const run = async (hostSlug: string, csvPath: string, options: RunOptions) => {
+  // Load host
+  const host = await models.Collective.findOne({
+    where: { slug: hostSlug },
+    include: [{ association: 'accountingCategories', required: false }],
+  });
+
+  if (!host) {
+    throw new Error(`Host ${hostSlug} not found`);
+  }
+
+  // Load CSV content
+  const rawContent = await loadFileContents(csvPath);
+  const parsedContent = parse(rawContent, {
+    delimiter: ',',
+    columns: true,
+    skip_empty_lines: true, // eslint-disable-line camelcase
+  });
+
+  // Iterate over rows
+  for (const row of parsedContent) {
+    const shortTransactionGroup = row['Short Group ID'];
+    const expenseAccountingCategoryCode = row['EXP CODE'];
+    if (!shortTransactionGroup || !expenseAccountingCategoryCode) {
+      if (options.verbose) {
+        console.error(`Missing data for row ${JSON.stringify(row)}`);
+      }
+      continue;
+    }
+
+    const expense = await models.Expense.findOne({
+      include: [
+        {
+          model: models.Transaction,
+          attributes: [],
+          required: true,
+          where: {
+            type: 'DEBIT',
+            kind: 'EXPENSE',
+            HostCollectiveId: host.id,
+            [Op.and]: [
+              sequelize.where(sequelize.cast(sequelize.col('TransactionGroup'), 'text'), {
+                [Op.startsWith]: shortTransactionGroup,
+              }),
+            ],
+          },
+        },
+        {
+          association: 'accountingCategory',
+          required: false,
+        },
+      ],
+    });
+
+    if (!expense) {
+      console.error(`Expense not found for short transaction group ${shortTransactionGroup}`);
+    } else if (expense.HostCollectiveId && expense.HostCollectiveId !== host.id) {
+      console.error(`Expense ${expense.id} is not associated with host ${hostSlug}`);
+    } else if (expenseAccountingCategoryCode === expense.accountingCategory?.code) {
+      if (options.verbose) {
+        console.log(`Expense ${expense.id} already has the correct accounting category`);
+      }
+    } else {
+      const newCategory = host.accountingCategories.find(({ code }) => code === expenseAccountingCategoryCode);
```

Comment:
This could also be null.

Reply:
Good catch, OCF is supposed to have a perfect match on their categories but that might not be the case for other hosts.